File: TypeScript/src/wrong.ts
import { Action } from "./action";
import { Base } from "./base";
import { Categories } from "./categories";
import { CurrentPlayer } from "./current_player";
import { PenaltyBox } from "./penalty_box";
import { Places } from "./places";
import { PlayerNames } from "./player_names";
import { Purses } from "./purses";
import { QuestionsByCategory } from "./questions_by_category";

export class Wrong extends Base implements Action {

    public constructor(
        currentPlayer: CurrentPlayer,
        names: PlayerNames,
        places: Places,
        purses: Purses,
        penaltyBox: PenaltyBox,
        categories: Categories,
        questions: QuestionsByCategory,
        isGettingOutOfPenaltyBox: boolean[]) {
        super(currentPlayer,
            names,
            places,
            purses,
            penaltyBox,
            categories,
            questions,
            isGettingOutOfPenaltyBox);
    }

    public execute(): boolean {
        this.println();

        this.penaltyBox.set(this.currentPlayer.get(), true);

        this.currentPlayer.inc();
        return true;
    }

    private println(): void {
        console.log("Question was incorrectly answered");
        this.penaltyBox.println();
    }
}

File: Java/src/main/java/com/adaptionsoft/games/uglytrivia/Wrong.java
package com.adaptionsoft.games.uglytrivia;

public class Wrong extends Base implements Action {

    public Wrong(CurrentPlayer currentPlayer,
            PlayerNames names,
            Places places,
            Purses purses,
            PenaltyBox penaltyBox,
            Categories categories,
            QuestionsByCategory questions,
            boolean[] isGettingOutOfPenaltyBox) {
        super(currentPlayer,
                names,
                places,
                purses,
                penaltyBox,
                categories,
                questions,
                isGettingOutOfPenaltyBox);
    }

    @Override
    public boolean execute() {
        println();

        penaltyBox.set(currentPlayer.get(), true);

        currentPlayer.inc();
        return true;
    }

    private void println() {
        System.out.println("Question was incorrectly answered");
        penaltyBox.println();
    }
}
